tests/worker/test_workflow.py

```diff
@@ -602,27 +667,67 @@ async def started() -> bool:
 
 @workflow.defn
 class CancelChildWorkflow:
+    def __init__(self) -> None:
+        self._ready = False
+
     @workflow.run
-    async def run(self, child_id: str) -> None:
-        handle = await workflow.start_child_workflow(LongSleepWorkflow.run, id=child_id)
-        await asyncio.sleep(0.1)
-        handle.cancel()
-        await handle
+    async def run(self, use_execute: bool) -> None:
+        if use_execute:
+            self._task = asyncio.create_task(
+                workflow.execute_child_workflow(
+                    LongSleepWorkflow.run, id=f"{workflow.info().workflow_id}_child"
+                )
+            )
+        else:
+            self._task = await workflow.start_child_workflow(
+                LongSleepWorkflow.run, id=f"{workflow.info().workflow_id}_child"
+            )
+        self._ready = True
+        await self._task
+
+    @workflow.query
+    def ready(self) -> bool:
+        return self._ready
+
+    @workflow.signal
+    async def cancel_child(self) -> None:
+        self._task.cancel()
 
 
-async def test_workflow_cancel_child(client: Client):
+@pytest.mark.parametrize("use_execute", [True, False])
+async def test_workflow_cancel_child_started(client: Client, use_execute: bool):
     async with new_worker(client, CancelChildWorkflow, LongSleepWorkflow) as worker:
         with pytest.raises(WorkflowFailureError) as err:
-            await client.execute_workflow(
+            # Start workflow
+            handle = await client.start_workflow(
                 CancelChildWorkflow.run,
-                f"child-workflow-{uuid.uuid4()}",
+                use_execute,
                 id=f"workflow-{uuid.uuid4()}",
                 task_queue=worker.task_queue,
             )
+            # Wait until child started
+            async def child_started() -> bool:
+                return await handle.query(
+                    CancelChildWorkflow.ready
+                ) and await client.get_workflow_handle_for(
+                    LongSleepWorkflow.run, workflow_id=f"{handle.id}_child"
+                ).query(
+                    LongSleepWorkflow.started
+                )
+
+            await assert_eq_eventually(True, child_started)
+            # Send cancel signal and wait on the handle
+            await handle.signal(CancelChildWorkflow.cancel_child)
+            await handle.result()
         assert isinstance(err.value.cause, ChildWorkflowError)
```

Comment:
Pardon my laziness, remind me what `err.value` is here? Is it the Failure object?

Reply:
It's the Python `Exception`. Pytest wraps it, so `err` here is technically https://docs.pytest.org/en/7.1.x/reference/reference.html#exceptioninfo